homeassistant/components/frontend/www_static/polymer/more-infos/more-info-camera.html

```diff
@@ -0,0 +1,357 @@
+      to your &quot;configuration.yaml&quot; file.  To view all the available configuration
+      options check the comments at the top of the &quot;components/ftp.py&quot; file.
+    </p>
+
+    <div center horizontal layout>
+    <div flex>
+        &nbsp;
+      </div>
+      <div>
+        <paper-button class="colored" on-click="{{closeFtpCompConfigDialog}}">
+          <core-icon icon="check"></core-icon>
+          OK
+        </paper-button>
+      </div>
+    </div>
+  </div>
+
+</template>
+<script>
+
+  var authStore = window.hass.authStore;
+  var serviceActions = window.hass.serviceActions;
+  var componentStore = window.hass.componentStore;
+
+  Polymer({
+    stateObj: null,
+    isStreaming: false,
+    stillImage: null,
+    reloadTime: null,
+    dialogOpen: false,
+    errorCount: 0,
+    connectionFailed: false,
+    motionDetectionIcon: 'visibility',
+    configureFTP: false,
+    selected: 0,
+    recordingIcon: 'av:videocam',
+    ftpComponentLoaded: false,
+    configureFtpComponent: false,
+    hasTabs: true,
+    supportsMotion: false,
+
+    stateObjChanged: function(oldVal, newVal) {
+      if(newVal) {
+        this.motionDetectionIcon = (newVal.attributes['is_motion_detection_enabled']) ? 'visibility-off' : 'visibility';
+        this.supportsMotion = newVal.attributes['is_motion_detection_supported'];
+        this.recordingIcon = (newVal.attributes['is_recording']) ? 'av:videocam-off' : 'av:videocam';
+      }
+    },
+
+    toggleMotionDetection: function(ev) {
+      if(!this.ftpComponentLoaded) {
+        this.configureFtpComponent = true;
+        return;
+      }
+      else if(!this.stateObj.attributes['is_ftp_configured']) {
+        this.configureFTP = true;
+        return;
+      }
+
+      if(this.stateObj.attributes['is_motion_detection_enabled'] === true) {
+        this.callCameraService('motion_detection', 'off');
+      } else {
+        this.callCameraService('motion_detection', 'on');
+      }
+    },
+
+    closeFtpConfigDialog: function(ev) {
+      this.configureFTP = false;
+    },
+
+    closeFtpCompConfigDialog: function(ev) {
+      this.configureFtpComponent = false;
+    },
+
+    setFtpDetails: function(ev) {
+      serviceActions.callService("camera", "turn_on", {
+          entity_id: this.stateObj.entityId,
+          feature: 'configure_ftp'
+        }).then(function() {
+          this.configureFTP = false;
+        }.bind(this));
+    },
+
+    dialogOpenChanged: function(oldVal, newVal) {
+      this.ftpComponentLoaded = componentStore.loaded.contains('ftp');
+      if(newVal == true) {
+        this.configureFTP = false;
+        this.configureFtpComponent = false;
+        this.startImageStream();
+        this.selected = 0;
+        this.recordingIcon = 'av:videocam';
+      }
+      else {
+        this.stopImageStream();
+      }
+    },
+
+    startImageStream: function() {
+      this.$.camera_image.src = this.stateObj.attributes['stream_url'];
```

Comment:
Use an object variable instead of modifying the DOM directly:

```html
<img src="{{camera_image_src}}" />
```

```javascript
startImageStream: function() {
  this.camera_image_src = this.stateObj.attributes.stream_url;
}
```